ExampleMod/Content/Tiles/ExampleCrystal.cs

```diff
@@ -0,0 +1,116 @@
+            Main.tile[i, j].TileType = Type;
+            return false;
+        }
+    }
+
+    // We need a GlobalTile class to make sure our crystal can randomly spawn in the world
+	// The logic is contained in the RandomUpdate function
+    internal class ExampleCrystalGrow : GlobalTile
+    {
+        public override void RandomUpdate(int i, int j, int type)
+        {
+			// The crystal will grow only if: the world is in Hardmode, the tile is Stone, it's in the rock layer
+			// and with same chance as the vanilla Crystal Shards have of spawning.
+            if (Main.hardMode && type == TileID.Stone && j > Main.rockLayer && WorldGen.genRand.NextBool(110))       
+            {
+				// Randomly generate the direction the crystal spawns in
+                int coordinateChance = WorldGen.genRand.Next(4);
+                int additionalX = 0;
+                int additionalY = 0;
+				
+                if (coordinateChance == 0)
+                {
+                    additionalX = -1;
+                    additionalY = -1;
+                }
+                else if (coordinateChance == 1)
+                {
+                    additionalX = 1;
+                    additionalY = 1;
+                }
+				
+				// Checks if the position where the crystal should spawn doesn't already have a tile
+                if (!Main.tile[i + additionalX, j + additionalY].HasTile)
+                {
+					// Count how many crystals there are in the area
+                    int amountOfCrystals = 0;
+                    int tilesToCheck = 6;
+                    int tile = ModContent.TileType<ExampleCrystal>();
+					
+                    for (int k = i - tilesToCheck; k <= i + tilesToCheck; k++)
+                    {
+                        for (int l = j - tilesToCheck; l <= j + tilesToCheck; l++)
+                        {
+                            if (Main.tile[k, l].HasTile && Main.tile[k, l].TileType == tile)
+                            {
+                                amountOfCrystals++;
+                            }
+                        }
+                    }
+					
+					// If there are less than 2 crystals, we can proceed to spawn one
+                    if (amountOfCrystals < 2)
+                    {
+                        int style = (short)WorldGen.genRand.Next(1);
+                        if (WorldGen.PlaceTile(i + additionalX, j + additionalY, TileID.Crystals, true, style: style)) // Places crystal shards first, because hardcoded!
+                        {
+                            Main.tile[i + additionalX, j + additionalY].TileType = ModContent.TileType<ExampleCrystal>(); // ... if succesfully placed, then replace with our crystal
+                            Main.Map.Update(i + additionalX, j + additionalY, Main.Map[i + additionalX, j + additionalY].Light); // also don't forget to update it on map, so it won't appear as crystal shards
```

Comment:
I wonder what exactly is hardcoded here? It would help if you add the explanation of it in a comment. So why do we have to place a normal crystal first?

Reply:
I dont get it? What is hardcoded so that we have to place a normal crystal first and replace it, instead of just placing our crystal ?

Reply:
![изображение_2022-06-09_070044882](https://user-images.githubusercontent.com/65787578/172748090-cee8a9f8-1fba-4576-b644-777f8335d19d.png)
(im bad at explaining things)

Reply:
I don't really get it. You will definitely have to fully research the vanilla crystal and port that over with 100% certainty to Example Mod